### DISCUSSION TITLE
Can we separately set source language(language of input audio file) and target language(language of transcription output) ?

### DISCUSSION THREAD

**Discussion by Rahulvisio** [SYSTRAN/faster-whisper#484] [General]
We want to pass source language (input audio language) to avoid auto detection hoping that it will reduce time and we want to pass target language for transcription output, but  we are able to see only one language parameter (that is The language spoken in the audio)
Does it represents the input audio language or transcription language?
If it represents input audio language where does shall we pass the transcription language?
If it represents transcription language where does shall we pass the  input audio language?

Use Case :  For (say)   using it to transcribe  for any other language apart from english

Code Snippet Below:

def transcribe(
        self,
        audio: Union[str, BinaryIO, np.ndarray],
        **language: Optional[str] = None,**
        task: str = "transcribe",
        beam_size: int = 5,
        best_of: int = 5,
        patience: float = 1,
        length_penalty: float = 1,
        repetition_penalty: float = 1,
        no_repeat_ngram_size: int = 0,
        temperature: Union[float, List[float], Tuple[float, ...]] = [
            0.0,
            0.2,
            0.4,
            0.6,
            0.8,
            1.0,
        ],
        compression_ratio_threshold: Optional[float] = 2.4,
        log_prob_threshold: Optional[float] = -1.0,
        no_speech_threshold: Optional[float] = 0.6,
        condition_on_previous_text: bool = True,
        prompt_reset_on_temperature: float = 0.5,
        initial_prompt: Optional[Union[str, Iterable[int]]] = None,
        prefix: Optional[str] = None,
        suppress_blank: bool = True,
        suppress_tokens: Optional[List[int]] = [-1],
        without_timestamps: bool = False,
        max_initial_timestamp: float = 1.0,
        word_timestamps: bool = False,
        prepend_punctuations: str = "\"'“¿([{-",
        append_punctuations: str = "\"'.。,，!！?？:：”)]}、",
        vad_filter: bool = False,
        vad_parameters: Optional[Union[dict, VadOptions]] = None,
    ) -> Tuple[Iterable[Segment], TranscriptionInfo]:
        """Transcribes an input file.

        Arguments:
          audio: Path to the input file (or a file-like object), or the audio waveform.
          language: The language spoken in the audio. It should be a language code such
            as "en" or "fr". If not set, the language will be detected in the first 30 seconds
            of audio.

**Comment by guillaumekln** [comment]
> Does it represents the input audio language or transcription language?

It is the audio language, as described in the documentation you copied.

> If it represents input audio language where does shall we pass the transcription language?

This option does not exist for Whisper. But Whisper has a `task` parameter:

* `task="transcribe"`: transcribe the audio in the same language
* `task="translate"`: transcribe and translate the audio into English
